cas-server-support-spnego/src/main/java/org/jasig/cas/support/spnego/web/flow/client/LdapSpnegoKnownClientSystemsFilterAction.java

```diff
@@ -0,0 +1,166 @@
+    
+    /**
+     * Instantiates new action. Initializes the default attribute
+     * to be {@link #DEFAULT_SPNEGO_ATTRIBUTE}.
+     * @param connectionFactory the connection factory
+     * @param searchRequest the search request
+     */
+    public LdapSpnegoKnownClientSystemsFilterAction(@NotNull final ConnectionFactory connectionFactory,
+                                                    @NotNull final SearchRequest searchRequest) {
+        this(connectionFactory, searchRequest, DEFAULT_SPNEGO_ATTRIBUTE);
+    }
+
+    /**
+     * Instantiates a new action.
+     *
+     * @param connectionFactory the connection factory
+     * @param searchRequest the search request
+     * @param spnegoAttributeName the certificate revocation list attribute name
+     */
+    public LdapSpnegoKnownClientSystemsFilterAction(
+            @NotNull final ConnectionFactory connectionFactory,
+            @NotNull final SearchRequest searchRequest,
+            @NotNull final String spnegoAttributeName) {
+        this.connectionFactory = connectionFactory;
+        this.spnegoAttributeName = spnegoAttributeName;
+        this.searchRequest = searchRequest;
+    }
+
+    /**
+     * Create and open a connection to ldap
+     * via the given config and provider.
+     *
+     * @return the connection
+     * @throws LdapException the ldap exception
+     */
+    protected Connection createConnection() throws LdapException {
+        logger.debug("Establishing a connection...");
+        final Connection connection = this.connectionFactory.getConnection();
+        connection.open();
+        return connection;
+    }
+
+    @Override
+    protected boolean shouldDoSpnego() {
+        final boolean ipCheck = ipPatternCanBeChecked();
+        if(ipCheck && !ipPatternMatches()) {
+            return false;
+        }
+
+        return executeSearchForSpnegoAttribute();
+    }
+
+    /**
+     * Searches the ldap instance for the attribute value.
+     *
+     * @return the boolean
+     */
+    protected boolean executeSearchForSpnegoAttribute() {
+        Connection connection = null;
+
+
+        final String remoteHostName = getRemoteHostName();
+        try {
+            connection = createConnection();
+            final Operation searchOperation = new SearchOperation(connection);
+            this.searchRequest.getSearchFilter().setParameter(0, remoteHostName);
+            final Response<SearchResult> searchResult = searchOperation.execute(this.searchRequest);
+            if (searchResult.getResultCode() == ResultCode.SUCCESS) {
+                return processSpnegoAttribute(searchResult);
+            }
+            throw new RuntimeException("Failed to establish a connection ldap. "
+                    + searchResult.getMessage());
+
+        } catch (final LdapException e) {
+            logger.error(e.getMessage(), e);
+            throw new RuntimeException(e);
+        } finally {
+            if (connection != null) {
```

Comment:
Ah nice. I was looking for one, and turns out it's closeConnection(). Once we move to Ldaptive 1.1, we should be able to take advantage of AutoClosables